stylize.py

```diff
@@ -111,20 +116,39 @@ def print_progress(i, last=False):
         best = None
         with tf.Session() as sess:
             sess.run(tf.initialize_all_variables())
-            for i in range(iterations):
-                last_step = (i == iterations - 1)
-                print_progress(i, last=last_step)
-                train_step.run()
-
-                if (checkpoint_iterations and i % checkpoint_iterations == 0) or last_step:
+            if optimizer == 'adam':
+                for i in range(iterations):
+                    last_step = (i == iterations - 1)
+                    print_progress(i, last=last_step)
+                    train_step()
+
+                    if (checkpoint_iterations and i % checkpoint_iterations == 0) or last_step:
+                        this_loss = loss.eval()
+                        if this_loss < best_loss:
+                            best_loss = this_loss
+                            best = image.eval()
+                        yield (
+                            (None if last_step else i),
+                            vgg.unprocess(best.reshape(shape[1:]), mean_pixel)
+                        )
+            elif optimizer == 'lbfgs':
+                if not checkpoint_iterations:
```

Comment:
Yeah… ideally, we'd investigate what's going on, and then hopefully we can get away with a much smaller diff to achieve the same effect.